compiler/modules/front.syntax/ast.bal

```diff
@@ -455,14 +455,19 @@ public type TypeDefn record {|
 
 public type TypeDesc BuiltinTypeDesc|BinaryTypeDesc|ConstructorTypeDesc|TypeDescRef|SingletonTypeDesc|UnaryTypeDesc;
 
-public type ConstructorTypeDesc ListTypeDesc|MappingTypeDesc|FunctionTypeDesc|ErrorTypeDesc|XmlSequenceTypeDesc;
+public type ConstructorTypeDesc TupleTypeDesc|ArrayTypeDesc|MappingTypeDesc|FunctionTypeDesc|ErrorTypeDesc|XmlSequenceTypeDesc;
 
-public type ListTypeDesc record {|
-    // JBUG #32617 can't include PositionFields
-    Position startPos;
-    Position endPos;
+public type TupleTypeDesc record {|
+    *PositionFields;
     TypeDesc[] members;
-    TypeDesc rest;
+    TypeDesc? rest;
+    t:ListDefinition? defn = ();
+|};
+
+public type ArrayTypeDesc record {|
+    *PositionFields;
+    TypeDesc elementTypeDesc;
+    SimpleConstExpr?[] arrayLen = [];
```

Comment:
The name needs improving. I don't think `array` adds value, since this is ArrayTypeDesc. We also try to keep the name of array-valued field to be plural. Another quibble is that nil isn't a length.

I suggest `dimensions` following the spec https://ballerina.io/spec/lang/master/#section_5.4.1.1